importer/EsEs.py

```diff
@@ -0,0 +1,135 @@
+        else:
+            self.add_statement("heritage_status", heritage)
+
+    def set_adm_location(self):
+        adm_q = None
+        municip_dic = self.data_files["municipalities"]
+        province_dic = self.data_files["provinces"]
+        if utils.count_wikilinks(self.municipio) == 1:
+            adm_q = utils.q_from_first_wikilink("es", self.municipio)
+        else:
+            # sometimes they're in ''
+            municip_raw = utils.remove_markup(self.municipio)
+            municip_match = [x for x in municip_dic if x[
+                "itemLabel"] == municip_raw]
+            if len(municip_match) == 1:
+                adm_q = municip_match[0]["item"]
+        if not adm_q:
+            province_raw = self.provincia_iso.lower()
+            prov_match = [x for x in province_dic if x[
+                "itemLabel"].lower() == province_raw]
+            if len(prov_match) == 1:
+                adm_q = prov_match[0]["item"]
+
+        if adm_q:
+            self.add_statement("located_adm", adm_q)
+        else:
+            self.add_to_report("municipio", self.municipio, "located_adm")
+
+    def set_special_is(self):
+        is_dic = self.data_files["tipobic"]["mappings"]
+        if self.has_non_empty_attribute("tipobic"):
+            special_raw = self.tipobic.lower()
+            special_match = [is_dic[x]
+                             for x in is_dic if x.lower() == special_raw]
+            if len(special_match) == 1:
+                self.remove_statement("is")
+                special_types = special_match[0]["items"]
+                for s in special_types:
+                    self.add_statement("is", s)
+
+    def update_labels(self):
+        spanish = utils.remove_markup(self.nombre)
+        self.add_label("es", spanish)
+
+    def update_descriptions(self):
+        desc_dic = {"es": "Bien de Interés Cultural",
+                    "en": "cultural property in Spain"}
+        for lg in desc_dic:
+            self.add_description(lg, desc_dic[lg])
+
+    def set_location(self):
+        if self.has_non_empty_attribute("lugar"):
+            if utils.count_wikilinks(self.lugar) == 1:
+                adm_loc = self.wd_item["statements"]["P131"][0]["value"]
+                loc_q = utils.q_from_first_wikilink("es", self.lugar)
+                if loc_q != adm_loc:
+                    self.add_statement("location", loc_q)
+            else:
+                self.add_to_report("lugar", self.lugar, "location")
+
+    def set_heritage_id(self):
+        self.add_statement("bien_de_interes", str(self.bic))
+
+    def set_monuments_all_id(self):
+        """Map which column name in specific table to  ID in monuments_all."""
+        self.monuments_all_id = self.bic
+
+    def exists_with_monument_article(self, language):
+        return super().exists_with_monument_article("es", "monument_article")
+
+    def __init__(self, db_row_dict, mapping, data_files, existing, repository):
```

Comment:
It looks like the monuments database defaults to `bic="0"` when no value is given. So we probably want to filter out all of those. I also spotted one with `bic="-"`.

Maybe filter out all with `bic in ("", "0", "-")`

Reply:
Added a check so that those are not uploaded. Also added "s/n" to the list.